packages/framework/src/Model/Article/Elasticsearch/ArticleIndex.php

```diff
@@ -32,10 +33,18 @@ public function getTotalCount(int $domainId): int
      * @param int $domainId
      * @param int $lastProcessedId
      * @param int $batchSize
+     * @param string[] $fields
      * @return array
      */
-    public function getExportDataForBatch(int $domainId, int $lastProcessedId, int $batchSize): array
-    {
+    public function getExportDataForBatch(
+        int $domainId,
+        int $lastProcessedId,
+        int $batchSize,
+        array $fields = [],
```

Comment:
It seems nice but I am afraid it would require too much additional effort that would postpone this task.

Also, with the current state, you could probably implement scopes for articles easier on a project. (I am not sure anyone would need that so maybe this is not a valid argument).

I am open to discussing it and creating a separate task for this but it would probably not get the highest priority anytime soon :shrug: